protocol/v1/qbft/instance/change_round.go

```diff
@@ -104,26 +105,22 @@ func (i *Instance) uponChangeRoundFullQuorum() pipelines.SignedMessagePipeline {
 				return
 			}
 
-			notPrepared, highest, e := i.HighestPrepared(signedMessage.Message.Round)
+			prepared, highest, e := i.HighestPrepared(i.State().GetRound())
```

Comment:
Spec is usin the msg round and not the state 

qbft/round_change.go:30

Reply:
@nivBlox Spec has these lines:
```go
noPrevProposal := state.ProposalAcceptedForCurrentRound == nil && state.Round == newRound
prevProposal := state.ProposalAcceptedForCurrentRound != nil && newRound > state.Round
```
, where `state` is `i.State` (instance state) and `newRound` is `signedRoundChange.Message.Round` (`signedRoundChange` is the message for which `uponRoundChange` is called). 

Therefore, you're right that we need to pass message round into `HighestPrepared`, but we need to get messages inside `HighestPrepared` according to state round, and the lines above should be rewritten to these ones:
```go
noPrevProposal := i.State().GetProposalAcceptedForCurrentRound() == nil && i.State().GetRound() == round
prevProposal := i.State().GetProposalAcceptedForCurrentRound() != nil && round > i.State().GetRound()
```
Thanks

Reply:
ok got it